components/accountability/AccountabilityHeatmap.tsx

```diff
@@ -0,0 +1,231 @@
+import {
+  eachDayOfInterval,
+  endOfYear,
+  format,
+  getDay,
+  startOfWeek,
+  startOfYear,
+} from "date-fns";
+import React, { useMemo } from "react";
+
+interface Entry {
+  creationDate: string;
+  isCompleted: boolean | null;
+}
+
+interface AccountabilityHeatmapProps {
+  entries: Entry[];
+  year?: number;
+  cellSize?: number;
+  cellGap?: number;
+  colors?: {
+    completed: string;
+    notCompleted: string;
+    noData: string;
+    empty: string;
+  };
+}
+
+const AccountabilityHeatmap: React.FC<AccountabilityHeatmapProps> = ({
+  entries,
+  year = new Date().getFullYear(),
+  cellSize = 16,
+  cellGap = 2,
+  colors = {
+    completed: "rgb(241, 135, 1)", // Skillfy Orange
+    notCompleted: "rgb(17 24 39)", // Skillfy Black
+    noData: "rgb(215, 215, 215)", // Grey
+    empty: "transparent", // Transparent
+  },
+}) => {
+  // Constants
+  const WEEK_DAYS = 7;
+  const MONTH_NAMES = [
+    "Jan",
+    "Feb",
+    "Mar",
+    "Apr",
+    "May",
+    "Jun",
+    "Jul",
+    "Aug",
+    "Sep",
+    "Oct",
+    "Nov",
+    "Dec",
+  ];
+  const DATE_FORMAT = "yyyy-MM-dd";
+
+  /**
+   * Generate all dates of the year, starting from the first Sunday before or on Jan 1.
+   */
+  const startDate = useMemo(
```

Comment:
do you really need to use useMemo everywhere? use memo is only helpful when you are trying to prevent expensive rendering/calculations from happening.

all of these date helper functions are pretty fast and quick. and the year values and the startDate, endData, aren't really going to change often, you're not saving much perf wise, and you're making this file harder to read.